util/release/main.go

```diff
@@ -0,0 +1,222 @@
+		log.Fatal("Error: failed to find protos in " + showcaseDir())
+	}
+
+	for _, f := range files {
+		execute("cp", f, protoDest)
+	}
+}
+
+func compileDescriptors() {
+	// Check if protoc is installed.
+	if err := exec.Command("protoc", "--version").Run(); err != nil {
+		log.Fatal("Error: 'protoc' is expected to be installed on the path.")
+	}
+
+	// Compile protos
+	protoDest := filepath.Join(
+		showcaseDir(),
+		"tmp",
+		"api-common-protos",
+		"google",
+		"showcase",
+		"v1alpha2")
+
+	files, err := filepath.Glob(filepath.Join(protoDest, "*.proto"))
+	if err != nil {
+		log.Fatal("Error: failed to find protos in " + protoDest)
+	}
+	command := []string{
+		"protoc",
+		"--proto_path=" + filepath.Join(showcaseDir(), "tmp", "api-common-protos"),
+		"--include_imports",
+		"--include_source_info",
+		"-o" + filepath.Join(showcaseDir(), "dist", fmt.Sprintf("gapic-showcase-%s.desc", version())),
+	}
+	execute(append(command, files...)...)
+}
+
+func tarProtos() {
+	tmpDir := filepath.Join(showcaseDir(), "tmp")
+	stagingDir := filepath.Join(tmpDir, fmt.Sprintf("gapic-showcase-v1alpha2-%s-protos", version()))
+	output := filepath.Join(showcaseDir(), "dist", fmt.Sprintf("gapic-showcase-v1alpha2-%s-protos.tar.gz", version()))
+
+	os.MkdirAll(stagingDir, 0755)
+	execute("cp", "-r", filepath.Join(tmpDir, "api-common-protos", "google"), stagingDir)
+
+	os.Chdir(tmpDir)
+	defer os.Chdir(showcaseDir())
+	execute("tar", "-zcf", output, fmt.Sprintf("gapic-showcase-v1alpha2-%s-protos", version()))
+}
+
+func compileBinaries() {
+	stagingDir := filepath.Join(showcaseDir(), "tmp", "x")
+	outputDir := filepath.Join(showcaseDir(), "dist")
+	execute("go", "get", "github.com/mitchellh/gox")
+
+	// This is a windows dependency that is not implicitly got since we only get
+	// the linux dependencies
+	execute("go", "get", "github.com/inconshreveable/mousetrap")
+
+	osArchs := []string{
+		"windows/amd64",
+		"linux/amd64",
+		"darwin/amd64",
+		"linux/arm",
+	}
+	for _, osArch := range osArchs {
+		execute(
+			"gox",
+			fmt.Sprintf("-osarch=%s", osArch),
+			"-output",
+			filepath.Join(stagingDir, fmt.Sprintf("gapic-showcase-v1alpha2-%s-{{.OS}}-{{.Arch}}/gapic-showcase", version())),
+			"github.com/googleapis/gapic-showcase")
+	}
+
+	dirs, _ := filepath.Glob(filepath.Join(stagingDir, "*"))
+	for _, dir := range dirs {
+		os.Chdir(dir)
+		// The windows binaries are suffixed with '.exe'. This allows us to create
+		// tarballs of the executables whether or not they contain a suffix.
+		files, _ := filepath.Glob(filepath.Join(dir, "gapic-showcase*"))
+		execute(
+			"tar",
+			"-zcf",
+			filepath.Join(outputDir, filepath.Base(dir)+".tar.gz"),
+			filepath.Base(files[0]))
+	}
+	os.Chdir(showcaseDir())
+}
+
+func execute(args ...string) {
+	log.Print("Executing: ", strings.Join(args, " "))
+	if output, err := exec.Command(args[0], args[1:]...).CombinedOutput(); err != nil {
+		log.Fatalf("%s", output)
+	}
+}
+
+var sDirMu sync.Mutex = sync.Mutex{}
```

Comment:
I think `sync.Once` is more appropriate here. Make the function update `showcaseDirMemo`, then 
```go
onceObj.Do(f)
return showcaseDirMemo
```

Reply:
Done. Although, I did it using a closure. IDK if thats idiomatic in go.